tests/CPT_test.py

```diff
@@ -0,0 +1,332 @@
+    model = get_peft_model(base_model, config_random)
+    emb = model.prompt_encoder.default.embedding.weight.data.clone().detach()
+    training_args = TrainingArguments(
+        output_dir="./results",
+        per_device_train_batch_size=1,
+        num_train_epochs=2,
+        remove_unused_columns=False,
+        save_strategy="no",
+        logging_steps=1,
+    )
+
+    train_dataset = dataset(sst_data["train"], global_tokenizer)
+
+    trainer = Trainer(model=model, args=training_args, train_dataset=train_dataset, data_collator=collator)
+
+    try:
+        trainer.train()
+    except Exception as e:
+        pytest.fail(f"Training failed with error: {e}")
+
+    assert torch.all(model.prompt_encoder.default.embedding.weight.data.clone().detach().cpu() == emb.cpu())
+
+    model.prompt_encoder.default.projection()
+    delta_emb = model.prompt_encoder.default.delta_embedding.weight.data.clone().detach()
+    norm_delta = delta_emb.norm(dim=1).cpu()
+    epsilon = model.prompt_encoder.default.get_epsilon().cpu()
+    assert torch.all(norm_delta <= epsilon)
+
+
+def test_model_training_text(sst_data, global_tokenizer, collator, config_text):
+    """Perform a short training run to verify the model and data integration."""
+
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+    model = get_peft_model(base_model, config_text)
+    emb = model.prompt_encoder.default.embedding.weight.data.clone().detach()
+
+    training_args = TrainingArguments(
+        output_dir="./results",
+        per_device_train_batch_size=1,
+        num_train_epochs=2,
+        remove_unused_columns=False,
+        save_strategy="no",
+        logging_steps=1,
+    )
+
+    train_dataset = dataset(sst_data["train"], global_tokenizer)
+
+    trainer = Trainer(model=model, args=training_args, train_dataset=train_dataset, data_collator=collator)
+
+    try:
+        trainer.train()
+    except Exception as e:
+        pytest.fail(f"Training failed with error: {e}")
+
+    assert torch.all(model.prompt_encoder.default.embedding.weight.data.clone().detach().cpu() == emb.cpu())
+
+    delta_emb = model.prompt_encoder.default.delta_embedding.weight.data.clone().detach()
+    norm_delta = delta_emb.norm(dim=1).cpu()
+    CPT_tokens_type_mask = torch.Tensor(config_text.CPT_tokens_type_mask).long()
+    non_label_idx = (CPT_tokens_type_mask == 1) | (CPT_tokens_type_mask == 2) | (CPT_tokens_type_mask == 3)
+    assert torch.all((norm_delta > 0) == non_label_idx)
+
+    model.prompt_encoder.default.projection()
+    delta_emb = model.prompt_encoder.default.delta_embedding.weight.data.clone().detach()
+    norm_delta = delta_emb.norm(dim=1).cpu()
+    epsilon = model.prompt_encoder.default.get_epsilon().cpu()
+    assert torch.all(norm_delta <= epsilon)
+    assert torch.all((norm_delta == 0) == (~non_label_idx))
+
+
+def test_model_batch_training_text(sst_data, global_tokenizer, collator, config_text):
```

Comment:
Same argument as for the tests above. Also, what exactly is different in this test, just the batch size? Why is it important to test batch size 1 and 2?